File: clojure/math/sieve.clj
;; Print the primes below 100, sieved by removing each prime's multiples from the candidates.
;; Run: clojure -M sieve.clj

(require '[clojure.string :as string])

(defn sieve [candidates]
  (if (empty? candidates)
    []
    (let [prime (first candidates)]
      (cons prime (sieve (remove #(zero? (mod % prime)) (rest candidates)))))))

(println (string/join " " (sieve (range 2 100))))

File: elixir/math/sieve.exs
# Print the primes below 100, sieved by rejecting each prime's multiples from the candidates.
# Run: elixir sieve.exs

defmodule Sieve do
  def primes([]), do: []

  def primes([prime | rest]) do
    [prime | primes(Enum.reject(rest, &(rem(&1, prime) == 0)))]
  end
end

2..99
|> Enum.to_list()
|> Sieve.primes()
|> Enum.join(" ")
|> IO.puts()
